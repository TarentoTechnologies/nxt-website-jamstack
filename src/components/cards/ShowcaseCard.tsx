import { Link } from "gatsby";
import { GatsbyImage, getImage } from "gatsby-plugin-image";
import ReactHtmlParser from "react-html-parser";

import forwardArrow from "../../../static/icons/arrow_forward_black.svg";
import {
  bgGlaucousGreen,
  bgWhite,
  black,
} from "../../styles/style-guide/ColorGuide.module.css";
import {
  cta1,
  heading700,
  subText2,
} from "../../styles/style-guide/Typography.module.css";
import { ctaIcon } from "../cards/PortfolioCard.module.css";
import {
  ctaIconPos,
  imgContainer,
  showCaseCard,
  showCaseImg,
} from "./ShowcaseCard.module.css";

interface ShowcaseCardProps {
  imgSrc: any;
  imgAlt: string;
  title: string;
  description: any;
  ctaText: string;
  ctaLink: any;
}

export const ShowcaseCard = ({
  imgSrc,
  imgAlt,
  title,
  description,
  ctaText,
  ctaLink,
}: ShowcaseCardProps) => {
  const image: any = getImage(imgSrc);

  return (
    <div className={`container-fluid`}>
      <div className={`row ${showCaseCard} ${bgWhite}`}>
        {/* Left side */}
        {imgSrc && imgSrc !== null ? (
          <div
            className={`col-sm-12 col-md-12 col-lg-12 col-xl-6 p-0 m-0 object-fit-cover d-flex justify-content-center align-items-center ${imgContainer}`}
          >
            {/* <img
              src={imgSrc}
              alt={imgAlt}
              className={`${showCaseImg} img-fluid`}
            /> */}
            <GatsbyImage
              image={image}
              alt={imgAlt ? imgAlt : ""}
              className={`${showCaseImg} img-fluid`}
              data-aos="fade-up"
            />
          </div>
        ) : (
          <div
            className={`col-sm-12 col-md-12 col-lg-12 col-xl-6 p-0 m-0 object-fit-cover ${bgGlaucousGreen} ${imgContainer}`}
            data-aos="fade-up"
          >
            <img
              src={imgSrc}
              alt={imgAlt}
              className={`${showCaseImg} img-fluid`}
            />
          </div>
        )}

        {/* Right side */}
        <div className="col-sm-12 col-md-12 col-lg-12 col-xl-6 d-flex align-items-center">
          <div className="my-4 my-sm-4 my-md-4 my-lg-4 my-xl-0 mx-4">
            <h2 className={`${heading700}`} data-aos="fade-up">
              {title}
            </h2>
            <div className={`${subText2} ${black} py-3`} data-aos="fade-up">
              {/* {description} */}
              {ReactHtmlParser(description?.data?.childMarkdownRemark?.html)}
            </div>
            {/* CTA */}
            <div className="mt-3">
              <Link to={ctaLink}>
                <label
                  className={`${cta1} pe-auto cursorPointer`}
                  role="link"
                  data-aos="fade-up"
                >
                  {ctaText}
                  <span className={`${ctaIconPos}`}>
                    <img
                      className={`${ctaIcon} ps-2`}
                      src={forwardArrow}
                      alt={ctaText}
                      data-aos="fade-up"
                    />
                  </span>
                </label>
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
