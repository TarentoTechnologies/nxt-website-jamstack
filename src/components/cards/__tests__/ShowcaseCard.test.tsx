import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { ShowcaseCard } from "../ShowcaseCard";

const imgData = {
  localFile: {
    childImageSharp: {
      gatsbyImageData: {
        layout: "constrained",
        placeholder: {
          fallback:
            "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAJABQDASIAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAIDBf/EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAHLWq1MoH//xAAZEAACAwEAAAAAAAAAAAAAAAAAEQEQITH/2gAIAQEAAQUCWaIjlf/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQIBAT8BP//EABQQAQAAAAAAAAAAAAAAAAAAACD/2gAIAQEABj8CX//EABsQAAEEAwAAAAAAAAAAAAAAAAABEBFxMUFR/9oACAEBAAE/IdiTNC2JcMNP/9oADAMBAAIAAwAAABAwD//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8QP//EABcRAQADAAAAAAAAAAAAAAAAAAABESH/2gAIAQIBAT8QnVP/xAAcEAEAAQQDAAAAAAAAAAAAAAABEQAQITFhkbH/2gAIAQEAAT8QJlFDUvaDwnauVbHdv//Z",
        },
        images: {
          fallback: {
            src: "/static/c5ab777b81ede6691c6b6d65988b39cb/02c3d/features_1_6765d2890b.png",
            srcSet:
              "/static/c5ab777b81ede6691c6b6d65988b39cb/07064/features_1_6765d2890b.png 193w,\n/static/c5ab777b81ede6691c6b6d65988b39cb/35e06/features_1_6765d2890b.png 386w,\n/static/c5ab777b81ede6691c6b6d65988b39cb/02c3d/features_1_6765d2890b.png 772w",
            sizes: "(min-width: 772px) 772px, 100vw",
          },
          sources: [],
        },
        width: 772,
        height: 354,
      },
    },
  },
};

describe("Test if the ShowcaseCard component is working as expected", () => {
  it("<ShowcaseCard /> matches snapshot", () => {
    const component = render(
      <ShowcaseCard
        imgSrc={imgData}
        imgAlt={"img1"}
        title={"Bihar Museum"}
        description={"Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum"}
        ctaText={"Read More"}
        ctaLink={"/"}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<ShowcaseCard /> renders without exploding", () => {
    const component = render(
      <ShowcaseCard
        imgSrc={imgData}
        imgAlt={"Logo"}
        title={"Bihar Museum"}
        description={"Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum"}
        ctaText={"Read More"}
        ctaLink={"/"}
      />
    );
    expect(component.getByTestId("showcase-card")).toBeInTheDocument();
  });
});
