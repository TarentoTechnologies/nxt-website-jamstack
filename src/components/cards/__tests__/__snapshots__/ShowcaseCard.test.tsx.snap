// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the ShowcaseCard component is working as expected <ShowcaseCard /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid"
    data-testid="showcase-card"
  >
    <div
      class="row showCaseCard bgWhite"
    >
      <div
        class="col-sm-12 col-md-12 col-lg-12 col-xl-6 p-0 m-0 object-fit-cover d-flex justify-content-center align-items-center imgContainer"
      />
      <div
        class="col-sm-12 col-md-12 col-lg-12 col-xl-6 d-flex align-items-center"
      >
        <div
          class="my-4 my-sm-4 my-md-4 my-lg-4 my-xl-0 mx-4"
        >
          <h2
            class="heading700"
            data-aos="fade-up"
          >
            Bihar Museum
          </h2>
          <div
            class="subText2 black py-3"
            data-aos="fade-up"
          />
          <div
            class="mt-3"
          >
            <a
              href="/"
            >
              <label
                class="cta1 pe-auto cursorPointer"
                data-aos="fade-up"
                role="link"
              >
                Read More
                <span
                  class="ctaIconPos"
                >
                  <img
                    alt="Read More"
                    class="ctaIcon ps-2"
                    data-aos="fade-up"
                    src="test-file-stub"
                  />
                </span>
              </label>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
